sdk/src/wallet/operations/transaction/sign_transaction.rs

```diff
@@ -78,7 +78,10 @@ where
             .secret_manager
             .read()
             .await
-            .transaction_unlocks(prepared_transaction_data)
+            .transaction_unlocks(
+                prepared_transaction_data,
+                &self.client().get_protocol_parameters().await?,
+            )
```

Comment:
I'ld rather get the protocol parameters before we're taking the lock on the secret manager.